docs/src/main/asciidoc/security-openid-connect-multitenancy-concept.adoc

```diff
@@ -0,0 +1,317 @@
+
+* A `tenant identifier` which uniquely identifies the realm where the end user was authenticated / logged in. While you can technically use any means of sending the `tenant identifier` from the user's browser or other client-side application, it is convenient to insert it as a string within the context path of the service.
+* An `idToken` holding user information in the form of a JSON Web Token which contains a set of properties (called `claims` in OAuth2/OIDC vocabulary) for the authenticated user. Refer to https://jwt.io[JWT IO] for examples and  https://www.rfc-editor.org/rfc/rfc7519[RFC 7519] for the JWT specification.
+
+
+== The 1-minute overview of a multi-tenant application
+
+Multi-tenant applications can enable development teams to quickly provide and maintain functionality to many organisations without compromising security concerns. However, understanding and translating concepts and best practises from the OIDC world can be a daunting task, so this overview will illustrate the workings of a typical multi-tenant application to highlight the OIDC and security-related parts.
+
+.Two clients calling a multi-tenant application
+[caption="Figure 1: "]
+image::oidc-multitenancy-concept-1.svg[Multitenant Application]
+
+The image above shows 2 users from different organisations call a calendar service to find their respective organisations' festivities. For the sake of simplicity, we assume that the application serves RESTful data to a browser-based Single-Page Application (SPA).
+
+. Each user has logged in using their organisation's own single-sign ("SSO"), which is frequently called OIDC Authentication Provider (called `"OpenID Provider"`, `"Identity Provider"` or `"IdP"` in OIDC literature). Upon login, the SSO has produced an `IdToken` JWT with user data and delivered that to the user's web browser.
+. The two user will call the calendar service with _different resource paths_ within the service URL to reach the calendar service. The application uses the `tenant identifier` part of the URI to identify the caller's organisation. Should an authenticated user from organisation 1 accidentally use a resource path allocated to organisation 2, the call will yield an unauthenticated HTTP response.
+. The `IdToken` data retreived from the own organisation's SSO is normally sent to the multi-tenant application with each call as an HTTP header, called `"authorization:"`. In this respect, an IdToken works as a session cookie which contains user information.
+
+With every call, the multi-tenant application can translate the `tenant identifier` "org1" to imply that the user is logged in to the SSO of organisation 1, and "org2" to the SSO of organisation 2. The multi-tenant application can also verify that the user information in the `IdToken` is not changed, and extract the data within it.
```

Comment:
The actual token verification (signature, etc) is controlled by Quarkus, users can affect the way the verification is done but typically, it would be implicit for a typical user application :-)